### PR TITLE
Check units

### PR DESCRIPTION
**Pull request by JulReinhardt**
Check units methods (WIP) and add units to APS loader

### CODE REVIEW

File: creator/nx_creator_xpcs.py [hunk 1]
```diff
@@ -14,20 +15,20 @@
...
-# TODO: not defined in NeXus at this time
+#TODO: not defined in NeXus at this time
 # Need to create & propose this new Application Definition to NIAC
 
 
 class NXCreator:
     """
-
-    Write a NeXus file from the XPCS data
+    Write a NeXus file from the XPCS and SAXS data
```
Comment (prjemian) [inline creator/nx_creator_xpcs.py:24]: all one line, spell correction for `definition`

File: creator/nx_creator_xpcs.py [hunk 2]
```diff
@@ -78,6 +78,47 @@ def _create_dataset(self, group, name, value, **kwargs):
         ds.attrs["target"] = ds.name
         return ds
 
+    def _check_unit(self, name, expected, supplied):
+        """
```
Comment (prjemian) [inline creator/nx_creator_xpcs.py:82]: indentation should match here

File: creator/nx_creator_xpcs.py [hunk 3]
```diff
@@ -106,30 +146,30 @@ def create_entry_group(self,
...
 
-            # FIXME: Check NeXus structure: point to this group for default plot
-            #file.attrs["default"] = self.entry_group.name.split("/")[-1]
-            file.close()
 
     def create_xpcs_group(self,
                           g2: np.ndarray = None,
                           g2_unit: str = 'a.u',
                           g2_stderr: np.ndarray = None,
+                          g2_stderr_unit: str = 'a.u',
```
Comment (prjemian) [inline creator/nx_creator_xpcs.py:157]: addressing these units in #34

File: creator/nx_creator_xpcs.py [hunk 4]
```diff
@@ -155,32 +195,36 @@ def create_xpcs_group(self,
...
             # TODO do we really want an instrument group here or directly adding mask as a subentry?
             instrument_group = self._init_group(self.xpcs_group, "instrument", "NXdata")
             mask_group = self._init_group(instrument_group, "mask", "NXdata")
-            self._create_dataset(mask_group, "mask", mask, units="au")
-            self._create_dataset(mask_group, "dqmap", dqmap, units="au")
-            self._create_dataset(mask_group, "dqlist", dqlist, units="au")
-            self._create_dataset(mask_group, "dphilist", dphilist, units="au")
-            self._create_dataset(mask_group, "sqmap", sqmap, units="au")
-            file.close()
+            self._create_dataset(mask_group, "mask", mask)
```
Comment (prjemian) [inline creator/nx_creator_xpcs.py:214]: there will be a merge conflict due to #34

**Review by prjemian — CHANGES_REQUESTED** [review]
changes as noted

File: creator/nx_creator_xpcs.py [hunk 5]
```diff
@@ -78,6 +78,47 @@ def _create_dataset(self, group, name, value, **kwargs):
...
+            and pint if not used for the unit check.
+
+        :param : name of field
+        :param : expected units
+        :param : units string that was given
+        :return *bool*: `True` if units conversion is possible:
+        """
+
+        # catch arbitrary unit separately from pint --> point that out in documentation
+        if supplied == 'au' or 'a.u.' or 'a.u':
```
Comment (AbbyGi) [inline creator/nx_creator_xpcs.py:95]: I think this syntax should be:
```py
if supplied in ["au", "a.u.", "a.u"]:
```
The current implementation will default to True even if supplied doesn't match any of the options.

File: creator/nx_creator_xpcs.py (hunk 1 above)
Comment (prjemian) [reply creator/nx_creator_xpcs.py:24]: such as: `Write NeXus file from XPCS & SAXS data using NXxpcs & NXcansas definitions`

**Review by AbbyGi — CHANGES_REQUESTED** [review]
Minor syntax change

**Review by prjemian — APPROVED** [review]
LGTM

**Review by AbbyGi — APPROVED** [review]
Looks good

**Pull request merged**